build_tools/circle/build_push_doc.sh

```diff
@@ -160,7 +160,7 @@ ls -la
 # Finally, deploy the branch, but if it's a pull request or tag, don't!!
 if [[ ! -z ${CIRCLE_PULL_REQUEST} ]]; then
   echo "Will not deploy doc on pull request (${CIRCLE_PULL_REQUEST})"
-elif [[ ${CIRCLE_BRANCH} == "master" || ! -z ${CIRCLE_TAG} ]]; then
+elif [[ ${CIRCLE_BRANCH} == "master" || (! -z ${CIRCLE_TAG} && ${CIRCLE_TAG} =~ '^v?[0-9]+\.[0-9]+\.?[0-9]*?[a-zA-Z]+[0-9]*$') ]]; then
```

Comment:
And here

Reply:
Does this need to be

```
((! -z ${CIRCLE_TAG}) && (${CIRCLE_TAG} =~ '^v?[0-9]+\.[0-9]+\.?[0-9]*?[a-zA-Z]+[0-9]*$'))
```

? Just like being liberal with parens as I am wary of order of operations in this stuff. Also, maybe update regex depending on the comments above.

Reply:
I can update that for sure. Also, I am going to do a second pass after this PR. I think technically `! -z` should be replaced with `-n` ([source](https://github.com/koalaman/shellcheck/wiki/SC2236)), and we have a lot of occurrences of that.